stellargraph/layer/knowledge_graph.py

```diff
@@ -175,27 +172,66 @@ def rank_edges_against_all_nodes(model, test_data, known_edges_graph):
 
         num_nodes = known_edges_graph.number_of_nodes()
 
-        def ranks(pred, true_ilocs):
-            batch_size = len(true_ilocs)
+        def ranks(
```

Comment:
Could you add a docstring for `ranks`?

Also I think `ranks` can pulled out of `ComplEx` now - it doesn't seem have anything `ComplEx` specific and I think the only outer scope variables it uses are `knowledge_graph` and `num_nodes`.

Reply:
I'm generally resistant about adding too much documentation for scoped and private functions, because maintaining documentation is error-prone, and it just drifts. However, I will move it out of `ComplEx` to `_ranks_from_score_columns` (or something) and give it a doc string.